src/discord-cluster-manager/cogs/leaderboard_cog.py

```diff
@@ -285,43 +346,51 @@ async def get_leaderboard_submissions(
-            leaderboard_id = db.get_leaderboard(leaderboard_name)["id"]
-            if not leaderboard_id:
+        try:
+            with self.bot.leaderboard_db as db:
+                # TODO: query that gets leaderboard id given leaderboard name
+                leaderboard_id = db.get_leaderboard(leaderboard_name)["id"]
+                if not leaderboard_id:
+                    await interaction.response.send_message(
+                        f'Leaderboard "{leaderboard_name}" not found.', ephemeral=True
+                    )
+                    return
+
+                submissions = db.get_leaderboard_submissions(leaderboard_name)
+
+            if not submissions:
                 await interaction.response.send_message(
-                    "Leaderboard not found.", ephemeral=True
+                    f'No submissions found for "{leaderboard_name}".', ephemeral=True
                 )
                 return
 
-            # submissions = db.get_leaderboard_submissions(leaderboard_id)  # Add dtype
-            submissions = db.get_leaderboard_submissions(leaderboard_name)  # Add dtype
-
-        if not submissions:
-            await interaction.response.send_message(
-                "No submissions found.", ephemeral=True
+            # Create embed
+            embed = discord.Embed(
+                title=f'Leaderboard Submissions for "{leaderboard_name}"',
+                color=discord.Color.blue(),
             )
-            return
-
-        # Create embed
-        embed = discord.Embed(
-            title=f'Leaderboard Submissions for "{leaderboard_name}"',
-            color=discord.Color.blue(),
-        )
 
-        for submission in submissions:
-            user_id = await get_user_from_id(
-                submission["user_id"], interaction, self.bot
-            )
-            print("members", interaction.guild.members)
-            print(user_id)
+            for submission in submissions:
+                user_id = await get_user_from_id(
+                    submission["user_id"], interaction, self.bot
+                )
 
-            embed.add_field(
-                name=f"{user_id}: {submission['submission_name']}",
-                value=f"Submission speed: {submission['submission_score']}",
-                inline=False,
-            )
+                embed.add_field(
+                    name=f"{user_id}: {submission['submission_name']}",
+                    value=f"Submission speed: {submission['submission_score']}",
+                    inline=False,
+                )
 
-        await interaction.response.send_message(embed=embed)
+            await interaction.response.send_message(embed=embed)
+        except Exception as e:
+            logger.error(str(e))
+            if "'NoneType' object is not subscriptable" in str(e):
+                await interaction.response.send_message(
+                    f"The leaderboard '{leaderboard_name}' doesn't exist.",
```

Comment:
This would be weird because there was already a `db.get_leaderboard(...)` call above.